drf_haystack/serializers.py

```diff
@@ -131,24 +138,69 @@ def get_fields(self):
                     warnings.warn("Field '{field}' already exists in the field list. This *will* "
                                   "overwrite existing field '{field}'".format(field=field_name))
                 field_mapping[field_name] = declared_fields[field_name]
-
         return field_mapping
 
     def to_representation(self, instance):
         """
+        If we have a serializer mapping, use that.  Otherwise, use standard serializer behavior
+        """
+        serializers = getattr(self.Meta, "serializers", None)
+        if not serializers:
+            return self.default_to_representation(instance)
+
+        index = instance.searchindex
+        serializer_class = serializers.get(type(index), None)
+        if not serializer_class:
+            raise ImproperlyConfigured("Could not find serializer for %s in mapping" % index)
+        return serializer_class(context=self._context).to_representation(instance)
+
+    def default_to_representation(self, instance):
```

Comment:
Yeah, I agree modularity is a good thing, we just have to make sure we don't confuse people further by adding too many (new) steps in an already familiar chain.
Most people using drf already expects the `to_representation` method to do a certain thing, so we need to make our intent clear if we're heading down a too modular path.

So I thinks something like this would be good

    def to_representation(self, instance):
        if we_have_serializer_mapping:
            return self.mulit_serializer_representation(instance)   
        ...
        default code here
        ....

But, I'm not going to force this through, so if you think the modular approach is better, go for it ;)

As for the `highlighted` fields, I believe it's pretty straight forward.